transaction/core/src/fog_hint.rs

```diff
@@ -128,43 +135,50 @@ impl FogHint {
     /// # Arguments
     /// * ingest_server_private_key
     /// * encrypted fog hint payload
+    /// * default plaintext
+    /// * initialized output FogHint
     ///
     /// # Returns
-    /// * (Fog hint, true) on success (Default Fog hint, false) otherwise
+    /// * Choice(1) on success Choice(0) otherwise
+    #[inline(never)]
     pub fn ct_decrypt(
         ingest_server_private_key: &RistrettoPrivate,
         ciphertext: &EncryptedFogHint,
-    ) -> (Self, bool) {
-        let mut default_plaintext = GenericArray::<
-            u8,
-            Diff<EncryptedFogHintSize, <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize>,
-        >::default();
-
+        output: &mut Self,
+    ) -> Choice {
         let mut rng = McRng::default();
+        let mut plaintext = Plaintext::default();
         let default_pubkey = RistrettoPublic::from_random(&mut rng);
 
-        default_plaintext.as_mut()[..RISTRETTO_PUBLIC_LEN]
-            .copy_from_slice(&default_pubkey.to_bytes());
+        plaintext.as_mut()[..RISTRETTO_PUBLIC_LEN].copy_from_slice(&default_pubkey.to_bytes());
 
-        let (plaintext, mut success) = match VersionedCryptoBox::default()
+        let (real_plaintext, mut success) = match VersionedCryptoBox::default()
             .decrypt_fixed_length(ingest_server_private_key, ciphertext.as_ref())
         {
-            Ok(real_plaintext) => (real_plaintext, true),
-            Err(_) => (default_plaintext, false),
+            Ok((result, real_plaintext)) => (Plaintext(real_plaintext), result),
+            Err(_) => (plaintext, false),
         };
 
+        let choice = Choice::from(success as u8);
+        plaintext.conditional_assign(&real_plaintext, choice);
```

Comment:
Maybe my own bias, but it's not immediately obvious why this is CT save for errors parsing the key (which is fine, bogus points are going to be visible from outside the enclave anyways), so I'd suggest moving the conditional assignment inside the match statement.

For future designs, it's probably advisable to cleanly separate the CT from non-CT parts by doing this kind of input validation up front (e.g. in `::new()`), to ensure that any internal APIs are CT-capable.